### PR TITLE
Move documentation of COM interop features in both NETFX and Core into .NET Guide.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
## Summary

Move some of the documentation of COM interoperability features that exist in both .NET Framework and .NET Core into the .NET Guide instead of the .NET Framework Guide.

Add tutorial on exposing .NET components to COM in .NET Core as a counterpart to the .NET Framework documentation. (Blocked on Preview 7 release)

### CODE REVIEW

File: docs/standard/native-interop/cominterop.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+title: COM Interop in .NET
+description: Learn how to interoperate with COM libraries in .NET.
+author: jkoritzinsky
+ms.author: jekoritz
+ms.date: 07/11/2019
+---
+
+# COM Interop in .NET
+
+The Component Object Model (COM) lets an object expose its functionality to other components and to host applications. To help enable users to interoperate with their existing code bases, the .NET Framework has always provided strong support for interoperating with COM libraries. In .NET Core 3.0, a large portion of this support has been added to .NET Core on Windows. The documentation here will explain how the common COM interop techonologies work and how you can utilize them to interoperate with your existing COM libraries.
```
Comment (rpetrusha) [inline docs/standard/native-interop/cominterop.md]:
```suggestion
The Component Object Model (COM) lets an object expose its functionality to other components and to host applications on Windows platforms. To help enable users to interoperate with their existing code bases, the .NET Framework has always provided strong support for interoperating with COM libraries. In .NET Core 3.0, a large portion of this support has been added to .NET Core on Windows. The documentation here will explain how the common COM interop techonologies work and how you can utilize them to interoperate with your existing COM libraries.
```
I thought that making it clear that this applies to Windows only in the first sentence is important.

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks for handling this migration of interop documentatation to the .NET Guide, @jkoritzinsky. I've left some comments and suggested changes.

**Comment by rpetrusha** [comment]
@jkoritzinsky, there are merge conflicts. Can you resolve them?

File: docs/core/native-interop/exposing-dotnet-components-to-com.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
+---
```
Comment (mairaw) [inline docs/core/native-interop/exposing-dotnet-components-to-com.md]: File name: for SEO and general guidelines (https://review.docs.microsoft.com/en-us/help/contribute/file-path-guidelines?branch=master#name-your-files)
I think the file name of this article should be expose-components-to-com.md (dotnet is already part of the path and shouldn't be repeated if possible)

File: .openpublishing.redirection.json [hunk 3]
```diff
@@ -1358,6 +1358,31 @@
...
+            "redirect_document_id": true
+        },
+        {
+            "source_path" : "docs/framework/interop/runtime-callable-wrapper.md",
+            "redirect_url": "/dotnet/standard/native-interop/runtime-callable-wrapper",
+            "redirect_document_id": true
+        },
+        {
+            "source_path" : "docs/framework/interop/applying-interop-attributes.md",
+            "redirect_url": "/dotnet/standard/native-interop/applying-interop-attributes",
```
Comment (mairaw) [inline .openpublishing.redirection.json]: I'd the opportunity to fix some SEO and naming issues here as well

```suggestion
            "redirect_url": "/dotnet/standard/native-interop/apply-attributes",
```

File: .openpublishing.redirection.json [hunk 4]
```diff
@@ -1358,6 +1358,31 @@
...
+            "redirect_document_id": true
+        },
+        {
+            "source_path" : "docs/framework/interop/qualifying-net-types-for-introspection.md",
+            "redirect_url": "/dotnet/standard/native-interop/qualifying-net-types-for-introspection",
```
Comment (mairaw) [inline .openpublishing.redirection.json]:
```suggestion
            "redirect_url": "/dotnet/standard/native-interop/qualify-types-for-introspection",
```

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
The builds look really good, @jkoritzinsky. I'll merge this now.

**Pull request merged**